### PR TITLE
fix(hooks): remove stray '?' that made every 'yarn <anything>' fire tmux reminder

### LINKED ISSUE #2514: bug(hooks/pre-bash-tmux-reminder): misplaced `?` makes every `yarn <anything>` trigger the tmux reminder
### Summary

The tmux-reminder hook uses a regex where the `?` on the yarn subcommand group is misplaced. As written, the yarn branch of the alternation matches every command that starts with `yarn ` (including `yarn build`, `yarn dev`, `yarn add`, `yarn workspaces …`) — not just `yarn install` / `yarn test` as intended by the sibling branches.

### Location

`scripts/hooks/pre-bash-tmux-reminder.js:16`

```js
/(npm (install|test)|pnpm (install|test)|yarn (install|test)?|bun (install|test)|cargo build|make\b|docker\b|pytest|vitest|playwright)/
```

The `?` after `(install|test)` applies only to the yarn subcommand group, so:

- `yarn (install|test)?`  =  `yarn ` followed by optional `install|test`  =  matches `yarn ` + anything.

Every other package-manager branch (`npm`, `pnpm`, `bun`) requires the subcommand — only yarn is over-broad.

### Expected

Only genuinely long-running commands like `yarn install` and `yarn test` (parity with `npm (install|test)`, `pnpm (install|test)`, `bun (install|test)`) should trigger the "Consider running in tmux" reminder.

### Actual

Every `yarn <anything>` call outside tmux surfaces the reminder in additional context. This spams the additional-context channel for very short commands like `yarn --version`, `yarn add foo`, `yarn workspaces list`, etc.

### Reproduction

```bash
node -e "
const {run} = require('/tmp/ECC/scripts/hooks/pre-bash-tmux-reminder.js');
delete process.env.TMUX;
for (const cmd of ['yarn install', 'yarn test', 'yarn add react', 'yarn build', 'yarn dev', 'yarn --version']) {
  const r = run(JSON.stringify({ tool_input: { command: cmd } }));
  console.log(cmd.padEnd(20), '->', typeof r === 'object' ? 'REMINDER FIRED' : 'no reminder');
}
"
```

All six lines print `REMINDER FIRED`, but only the first two should.

### Suggested fix

Drop the stray `?`:

```diff
-/(npm (install|test)|pnpm (install|test)|yarn (install|test)?|bun (install|test)|cargo build|make\b|docker\b|pytest|vitest|playwright)/
+/(npm (install|test)|pnpm (install|test)|yarn (install|test)|bun (install|test)|cargo build|make\b|docker\b|pytest|vitest|playwright)/
```

Severity: **Minor** — UX noise, not a crash.

### PR DESCRIPTION
**Pull request by thejesh23**
## What Changed

Single-character fix in the tmux-reminder matcher regex in `scripts/hooks/pre-bash-tmux-reminder.js`:

```diff
-/(npm (install|test)|pnpm (install|test)|yarn (install|test)?|bun (install|test)|cargo build|make\b|docker\b|pytest|vitest|playwright)/
+/(npm (install|test)|pnpm (install|test)|yarn (install|test)|bun (install|test)|cargo build|make\b|docker\b|pytest|vitest|playwright)/
```

## Why This Change

The `?` after `(install|test)` on the yarn branch made the subcommand optional, so the branch degraded to matching `yarn ` plus anything: `yarn add react`, `yarn build`, `yarn dev`, even `yarn --version` all fired the "Consider running in tmux" additional-context hint. Every other package-manager branch (`npm`, `pnpm`, `bun`) correctly requires the subcommand.

## Testing Done

- [x] Manual testing completed — verified locally against 14 cases (yarn install/test still fire; yarn add/build/dev/--version/bare `yarn` no longer do; npm/pnpm/bun/pytest behavior unchanged). Reproduction and expected output are in #2514.
- [x] Automated tests pass locally — no dedicated test file exists for this hook; the pattern change is minimal and its behavior is exercised at runtime by the hook dispatcher.
- [x] Edge cases considered and tested — cargo, make, docker, pytest, vitest, playwright all still match.

## Type of Change

- [x] `fix:` Bug fix

## Security & Quality Checklist

- [x] No secrets or API keys committed
- [x] JSON files validate cleanly (none touched)
- [x] Shell scripts pass shellcheck — none touched
- [x] No sensitive data exposed in logs or output
- [x] Follows conventional commits format

## If you changed dependencies or `package.json`

No dependency changes.

## If you added a skill, command, agent, hook, or CLI tool

Not applicable — this is a regex fix inside an existing hook script; no new surface added.

## Documentation

No documented behavior changed. The reminder now correctly matches the intent documented by the sibling branches of the same alternation.

## Linked issue

Fixes #2514

---

Related independent PRs opened alongside this one (different files, no conflicts):
- fix(llm/providers/claude): cache_control shape — #2512
- fix(ecc_dashboard): Refresh Data repopulates Rules and Commands — #2513

### CODE REVIEW

**Comment by thejesh23** [comment]
Addressed CodeRabbit review in `2790045`: added `tests/hooks/pre-bash-tmux-reminder.test.js` modeled on `tests/hooks/auto-tmux-dev.test.js` (auto-discovered by `tests/run-all.js` via the `tests/**/*.test.js` glob). 16 assertions cover:

- yarn: install/test **fire**, add/build/dev/--version/bare-`yarn` **do not fire**
- sibling PMs: npm install / pnpm test / bun install still fire; npm run dev does not
- pytest, cargo build still fire
- TMUX env-var bypass respected
- malformed JSON and missing command tolerated

Verified the tests actually regress the bug: reintroducing `yarn (install|test)?` fails 4 of the 5 yarn non-match assertions.

File: tests/hooks/pre-bash-tmux-reminder.test.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+const script = path.join(__dirname, '..', '..', 'scripts', 'hooks', 'pre-bash-tmux-reminder.js');
+
+function test(name, fn) {
+  try {
+    fn();
+    console.log(`  ✓ ${name}`);
+    return true;
+  } catch (err) {
+    console.log(`  ✗ ${name}`);
+    console.log(`    Error: ${err.message}`);
```
[the remark this answers is not in the document]
Comment (thejesh23) [reply tests/hooks/pre-bash-tmux-reminder.test.js]: Declining — every sibling hook test in this repo uses `console.log` (auto-tmux-dev.test.js: 8, bash-hook-dispatcher.test.js: 5, block-no-verify.test.js: several) because the project's own test runner (`tests/run-all.js`) is console-based and there is no Jest/Mocha dependency. Swapping frameworks for a bugfix PR is out of scope; following convention keeps the file consistent.

File: tests/hooks/pre-bash-tmux-reminder.test.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    return false;
+  }
+}
+
+function runScript(input) {
+  const env = { ...process.env };
+  delete env.TMUX;
```
[the remark this answers is not in the document]
Comment (thejesh23) [reply tests/hooks/pre-bash-tmux-reminder.test.js]: Applied in `e0ff619` — now `const { TMUX: _tmuxDropped, ...env } = process.env;`.

File: tests/hooks/pre-bash-tmux-reminder.test.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    input: typeof input === 'string' ? input : JSON.stringify(input),
+    timeout: 10000,
+    env,
+  });
+  return {
+    code: result.status || 0,
+    stdout: result.stdout || '',
+    stderr: result.stderr || '',
+  };
+}
```
[the remark this answers is not in the document]
Comment (thejesh23) [reply tests/hooks/pre-bash-tmux-reminder.test.js]: Applied in `e0ff619` — `invokeHook` and `runRaw` now throw explicitly on `result.error`, `result.signal`, or null/undefined `result.status` (no more `|| 0` masking spawn failures).

File: tests/hooks/pre-bash-tmux-reminder.test.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+  if (test('handles missing command field', () => {
+    const result = runScript({ tool_input: {} });
+    assert.strictEqual(result.code, 0);
+    assert.ok(!result.stdout.includes('Consider running in tmux'));
+  })) passed++; else failed++;
+
+  console.log(`\nResults: ${passed} passed, ${failed} failed\n`);
+  return failed === 0;
+}
```
[the remark this answers is not in the document]
Comment (thejesh23) [reply tests/hooks/pre-bash-tmux-reminder.test.js]: Applied in `e0ff619` — `runTests` is now 18 lines (well under the 50-line limit). Extracted per-group helpers: `runYarnTests`, `runSiblingPackageManagerTests`, `runOtherToolTests`, `runTmuxBypassTests`, `runEdgeCaseTests`. `runTests` is now a pure orchestrator.

**Comment by rjjelley** [comment]
Local review from Codex automation: focused fix for #2514 looks good. The regex change removes the stray optional marker, and the added hook test exits cleanly on this Windows host via the intended skip path. Recommendation: merge after non-Windows CI confirms the hook behavior. I cannot merge from this integration (`403` on merge permission).
